### ISSUE TITLE
Only accept functions in then/catch

### ISSUE THREAD

**Issue by friday** [xjamundx/eslint-plugin-promise#60]
Similar to: http://bluebirdjs.com/docs/warning-explanations.html

Examples:

```js
Promise.resolve(1)
  .then(Promise.resolve(2)) // This is ignored, probably not what the developer intended
  .then(console.log) // 1
```

```js
const errorHandler = console.warn;
Promise.reject('Catch me if you can').catch(errorHandler()) // should have been `errorHandler` without parenthesis 
```

**Comment by macklinu** [comment]
I'm not 100% sure if this is possible without exploring a little bit (fairly new to ESLint plugin development), but I think this would be a great thing to consider and weave into #47. 👍

**Comment by friday** [comment]
As I read #47, that one seems to be limited to the number of arguments, which I think is much easier. I'm also not sure this issue is actually possible since references or function calls could be valid if they return other functions, and I'm not sure how much can be done with static code analysis.

Good luck either way, and thank you for the effort!
